kolibri/plugins/coach/assets/src/views/create-exam-page/index.vue

```diff
@@ -69,35 +128,122 @@
         validateTitle: false,
         validateNum: false,
         searchInput: '',
+        loading: false,
+        exercisesSelected: [],
+        topicsSelected: [],
+        seed: '',
       };
     },
     components: {
       'ui-select': require('keen-ui/src/UiSelect'),
+      'ui-snackbar': require('keen-ui/src/UiSnackbar'),
+      'ui-snackbar-container': require('keen-ui/src/UiSnackbarContainer'),
       'icon-button': require('kolibri.coreVue.components.iconButton'),
       'textbox': require('kolibri.coreVue.components.textbox'),
+      'topic-row': require('./topic-row'),
+      'exercise-row': require('./exercise-row'),
+      'preview-new-exam-modal': require('./preview-new-exam-modal'),
     },
     computed: {
-
       titleInvalid() {
         return this.validateTitle ? !this.inputTitle : false;
       },
       numQuestionsInvalid() {
         return this.validateNum ?
           (this.inputNumQuestions < 1) || (this.inputNumQuestions > 50) : false;
       },
+      showPreviewNewExamModal() {
+        return this.modalShown === ExamModals.PREVIEW_NEW_EXAM;
+      },
+    },
+    methods: {
+      handleGoToTopic(topicId) {
+        this.loading = true;
+        this.fetchContent(this.currentChannel.id, topicId).then(
+          () => {
+            this.loading = false;
+          },
+          error => {}
+        );
+      },
+      handleAddTopicExercises(topicId, topicTitle) {
```

Comment:
`handle..TopicExercise` and `handle...Exercise` are nearly identical. The only difference I can tell is that the first updates `topicsSelected` and the second updates `exercisesSelected`.

It might be possible to combine into a single method with signature `f(id, title, isForTopic /* bool */)`, and use `isForTopic` to determine which array to update.

Also, you should feel free to use `Array.prototype.includes` since it's polyfilled on the client-side.